src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,965 @@
+                }
+            }
+            return false
+        } else if (property === Property.Prime) {
+            return isPrime(num)
+        } else if (property === Property.Negative) {
+            return num < 0n
+        } else if (property === Property.Even) {
+            return num % 2n === 0n
+        } else if (property === Property.Divisible_By_Three) {
+            return num % 3n === 0n
+        } else if (property === Property.Divisible_By_Four) {
+            return num % 4n === 0n
+        } else if (property === Property.Divisible_By_Five) {
+            return num % 5n === 0n
+        } else if (property === Property.Divisible_By_Six) {
+            return num % 6n === 0n
+        } else if (property === Property.Divisible_By_Seven) {
+            return num % 7n === 0n
+        } else if (property === Property.Divisible_By_Eight) {
+            return num % 8n === 0n
+        } else if (property === Property.Ends_With_One) {
+            return num % 10n === 1n
+        } else if (property === Property.Ends_With_Two) {
+            return num % 10n === 2n
+        } else if (property === Property.Ends_With_Three) {
+            return num % 10n === 3n
+        } else if (property === Property.Ends_With_Four) {
+            return num % 10n === 4n
+        } else if (property === Property.Ends_With_Five) {
+            return num % 10n === 5n
+        } else if (property === Property.Ends_With_Six) {
+            return num % 10n === 6n
+        } else if (property === Property.Ends_With_Seven) {
+            return num % 10n === 7n
+        } else if (property === Property.Ends_With_Eight) {
+            return num % 10n === 8n
+        } else if (property === Property.Ends_With_Nine) {
+            return num % 10n === 9n
+        } else if (property === Property.Ends_With_Zero) {
+            return num % 10n === 0n
+        } else if (property === Property.Sum_Of_Two_Squares) {
+            return isSumOfTwoSquares(num)
+        } else if (property === Property.Triangular_Number) {
+            return isPolygonal(num, 3n)
+        } else if (property === Property.Square_Number) {
+            return isPolygonal(num, 4n)
+        } else if (property === Property.Pentagonal_Number) {
+            return isPolygonal(num, 5n)
+        } else if (property === Property.Hexagonal_Number) {
+            return isPolygonal(num, 6n)
+        } else if (property === Property.Heptagonal_Number) {
+            return isPolygonal(num, 7n)
+        } else if (property === Property.Octagonal_Number) {
+            return isPolygonal(num, 8n)
+        } else if (property === Property.Abundant) {
+            return isAbundant(num)
+        } else if (property === Property.Perfect) {
+            return isPerfect(num)
+        } else if (property === Property.Deficient) {
+            return !isPerfect(num) && !isAbundant(num)
+        } else if (property === Property.Semi_Prime) {
+            return isSemiPrime(num)
+        }
```

Comment:
Does Javascript have the capability to create a dictionary or something like that to call a function from a list of functions based on a key (the value of variable ```property```)?  It would maybe save on this repetition?  This may be a question for @gwhitney or @liammulh too.

Reply:
Yes, I completely agree that if an enum for the different Property values is kept as the implementation here, then they should be implemented by an array of functions, so that you can just call `propertyIndicatorFunction[property](num)` rather than needing a very long sequence of if-else-ifs.

Reply:
Create an object like @gwhitney described and in doing so condense the code.